src/main/javassist/util/proxy/ProxyFactory.java

```diff
@@ -1203,6 +1209,24 @@ private static boolean isOverloaded(int index, Method[] methods) {
 
         return false;
     }
+    
+    private static boolean isBridgeAndOverloaded(Method meth, List methods) {
+        if (isBridge(meth)) {
+            int cnt = 0;
+            String name = meth.getName();
+            for (Object m: methods) {
+                if (m instanceof Map.Entry<?, ?>) {
+                    Method targetMethod = ((Map.Entry<String, Method>)m).getValue();
+                    if (isBridge(targetMethod) && name.equals(targetMethod.getName())) {
```

Comment:
I still think the problem is with isBridge, here we are applying the fix, isBridge returns true but it shouldn't if `name.equals(targetMethod.getName())`

if we add that check to isBridged won't it also solve the problem?

Reply:
As I have explained above, and check if a method is overloaded, extra parameters are needed: all the methods info.